File: web/src/catalog.test.ts
import { describe, expect, it } from 'vitest'
import { baseline, catalog, failures } from './catalog'

describe('shared reliability catalog', () => {
  it('contains one baseline and six failure experiments', () => {
    expect(baseline.index).toBe('00')
    expect(failures).toHaveLength(6)
    expect(catalog.scenarios).toHaveLength(7)
  })

  it('keeps recovery evidence complete and relations unique', () => {
    const ids = new Set<string>()
    const relations = new Set<string>()
    for (const scenario of catalog.scenarios) {
      expect(ids.has(scenario.id)).toBe(false)
      ids.add(scenario.id)
      expect(scenario.fundInvariant).toBeTruthy()
      expect(scenario.firstAction).toBeTruthy()
      expect(scenario.recoveryBasis).toBeTruthy()
      expect(scenario.goTest).toMatch(/^Test/)
      expect(scenario.steps.length).toBeGreaterThanOrEqual(4)
      if (scenario.kind === 'failure') {
        expect(scenario.failureSlug).toBeTruthy()
        expect(relations.has(scenario.failureSlug)).toBe(false)
        relations.add(scenario.failureSlug)
      }
    }
  })
})

File: scenarios/catalog_test.go
package scenarios

import "testing"

func TestCatalogHasBaselineAndSixFailures(t *testing.T) {
	catalog, err := Load()
	if err != nil {
		t.Fatal(err)
	}
	if len(catalog.Scenarios) != 7 {
		t.Fatalf("scenario count = %d, want 7", len(catalog.Scenarios))
	}
	ids := map[string]bool{}
	failures := 0
	for _, scenario := range catalog.Scenarios {
		if scenario.ID == "" || ids[scenario.ID] {
			t.Fatalf("invalid or duplicate scenario id %q", scenario.ID)
		}
		ids[scenario.ID] = true
		if len(scenario.Steps) < 4 || scenario.FundInvariant == "" || scenario.FirstAction == "" || scenario.RecoveryBasis == "" || scenario.GoTest == "" {
			t.Fatalf("scenario %s is missing recovery evidence", scenario.ID)
		}
		if scenario.Kind == "failure" {
			failures++
			if scenario.FailureSlug == "" {
				t.Fatalf("failure scenario %s has no site relation", scenario.ID)
			}
		}
	}
	if failures != 6 {
		t.Fatalf("failure count = %d, want 6", failures)
	}
}
